internal/server/server_test.go

```
@@ -499,6 +500,196 @@ func TestIsExpectedWSClose(t *testing.T) {
+	valkeyClient.PopulateStatuses(map[string]*store.EndpointStatus{
+		"chainX:ep1": {HasWS: true, HealthyWS: true},
+	})
+	srv := NewServer(cfg, valkeyClient, createTestConfig())
+
+	proxy := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_ = srv.defaultProxyWebSocket(w, r, upstreamWSURL)
+	}))
+	t.Cleanup(proxy.Close)
+
+	proxyWSURL := "ws" + strings.TrimPrefix(proxy.URL, "http")
+	client, _, err := websocket.DefaultDialer.Dial(proxyWSURL, nil)
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)